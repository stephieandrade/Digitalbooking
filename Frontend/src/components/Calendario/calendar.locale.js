const gregorian_en_lowercase = {
  name: "gregorian_en_lowercase",
  months: [
    ["Enero", "Enero"],
    ["Febrero", "Febrero"],
    ["Marzo", "Marzo"],
    ["Abril", "Abril"],
    ["Mayo", "Mayo"],
    ["Junio", "Junio"],
    ["Julio", "Julio"],
    ["Agosto", "Agosto"],
    ["Septiembre", "Septiembre"],
    ["Octubre", "Octubre"],
    ["Noviembre", "Noviembre"],
    ["Diciembre", "Diciembre"],
  ],
  weekDays: [
    ["S", "S"],
    ["D", "D"],
    ["L", "L"],
    ["M", "M"],
    ["M", "W"],
    ["J", "J"],
    ["V", "V"],
  ],
  digits: ["0", "1", "2", "3", "4", "5", "6", "7", "8", "9"],
  meridiems: [
    ["AM", "am"],
    ["PM", "pm"],
  ],
};

export default gregorian_en_lowercase;
