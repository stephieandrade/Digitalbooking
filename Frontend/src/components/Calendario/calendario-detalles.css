.detalles.rmdp-wrapper {
  display: flex;
  width: 100%;
  padding: 4% 0 4% 0;
  justify-content: center;
}
.detalles .rmdp-calendar {
  width: 100%;
  /* padding-left: 10%; */
  /* padding-right: 10%; */
}
.detalles .rmdp-day-picker {
  gap: 2%;
  width: 100%;
}
.detalles .rmdp-day {
  margin: 7px;
  width: 34px;
  height: 34px;
}
.detalles .rmdp-day.rmdp-selected span:not(.highlight) {
  color: black;
  background-color: var(--one-color);
}
.detalles .rmdp-day span,
.detalles .rmdp-header-values span,
.detalles .rmdp-week-day {
  font-size: 20px;
}
.detalles .rmdp-week-day {
  color: var(--one-color);
}
.detalles .sd {
  /* margin: 0 -30% 0 0; */
}

.detalles .rmdp-day.rmdp-disabled span {
  color: #bebebe;
  text-decoration: line-through;
}
.detalles .rmdp-day {
  cursor: default;
}

.detalles .rmdp-day:not(.rmdp-disabled):not(.rmdp-day-hidden) span:hover {
  background-color: initial;
  color: initial;
  cursor: default;
}

@media screen and (min-width: 768px) {
  .detalles.rmdp-wrapper {
    padding-bottom: 3%;
    width: 100%;
    padding-right: 0;
    padding-left: 0;
  }

  /* Espacio entre cada uno de los meses*/
  .detalles .rmdp-day-picker {
    gap: 4%;
  }

  /* margin entre los días */
  .detalles .rmdp-day {
    margin: 5px;
  }
  .detalles .rmdp-week {
    justify-content: space-around;
  }

  /* Fuente del texto de días */
  .detalles .rmdp-day span,
  .detalles .rmdp-header-values span,
  .detalles .rmdp-week-day {
    font-size: 20px;
    font-weight: 100;
  }
  .detalles .rmdp-week-day {
    padding: 3% 0 3% 0;
  }
  .detalles .rmdp-day {
    /* color: #545776; */
  }

  .detalles .rmdp-calendar {
    height: 330px;
  }

  /* .detalles .rmdp-container .custom-calendar.ep-arrow[direction="bottom"] {
    border-top: 1px solid bisque;
  }
   */
  /* .rmdp-container .custom-calendar.ep-arrow[direction="top"] {
    border-bottom: 1px solid bisque;
  } */

  /* .detalles .rmdp-header{
      padding-bottom: 6%;
  } */
}

@media screen and (min-width: 992px) {
  .detalles.rmdp-wrapper {
    width: fit-content;
    /* padding: 5% 25% 0 20%; */
    padding: 35px 55px;
  }

  .detalles .rmdp-calendar {
    width: 800px;
  }
  .detalles .rmdp-day-picker {
    gap: 6%;
    justify-content: center;
  }
  .detalles .rmdp-day {
    /* margin: 1% 2%; */
    margin: 5px 7px;
  }
  .detalles .rmdp-day.rmdp-selected span:not(.highlight) {
    width: 30px;
  }
}
