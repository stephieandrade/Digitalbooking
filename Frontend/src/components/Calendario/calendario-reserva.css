.reserva {
  width: 100%;
  display: flex;
}
.reserva.rmdp-wrapper {
  display: flex;
  width: 100%;
  justify-content: center;
}
.reserva .rmdp-calendar {
  width: 100%;
}
.reserva1 .rmdp-day.rmdp-range {
  background-color: var(--one-color);
  border-radius: 50%;
  color: #fff;
}

.reserva .rmdp-day.rmdp-range {
  background-color: #1dbeb4;
  box-shadow: 0 0 3px #1dbeb4;
  color: #fff;
  border-radius: 50%;
}

.reserva .rmdp-day.rmdp-disabled span {
  color: #bebebe;
  text-decoration: line-through;
}

.reserva .rmdp-day-picker {
  display: flex;
  justify-content: space-around;
  gap: 20px;
}
.reserva .rmdp-day {
  margin: 7px;
  width: 34px;
  height: 34px;
}
.reserva .rmdp-day.rmdp-selected span:not(.highlight) {
  color: black;
  background-color: var(--one-color);
}
.reserva .rmdp-day span,
.reserva .rmdp-header-values span,
.reserva .rmdp-week-day {
  font-size: 20px;
}
.reserva .rmdp-week-day {
  color: var(--one-color);
}

.reserva .rmdp-header {
  padding-top: 10px;
}
.reserva .rmdp-day.rmdp-today span {
  color: #fff;
  width: 28px;
  background-color: #7fdbff;
}
.reserva .rmdp-day.rmdp-selected span:not(.highlight) {
  background-color: #1dbeb4;
  box-shadow: 0 0 3px #1dbeb4;
  color: #fff;
  width: 28px;
}

@media (min-width: 768px) {
  */ .reserva .rmdp-calendar {
    width: 100%;
  }

  .reserva .rmdp-day-picker {
    display: flex;
    justify-content: space-around;
    gap: 20px;
  }

  .reserva .rmdp-header {
    padding-top: 10px;
  }
  .reserva .rmdp-day.rmdp-today span {
    color: var(--two-color);
    width: 28px;
  }
  .reserva .rmdp-day.rmdp-selected span:not(.highlight) {
    background-color: #1dbeb4;
    box-shadow: 0 0 3px #1dbeb4;
    color: #fff;
    width: 28px;
  }
}
/**--------------MODELO ADAPTATIVO PC---------------*/

@media (min-width: 992px) {
  .reserva.rmdp-wrapper {
    width: auto;
    justify-content: center;
  }
  .rmdp-wrapper.rmdp-shadow.reserva {
    width: 100%;
  }
}
