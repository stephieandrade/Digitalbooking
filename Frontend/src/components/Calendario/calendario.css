@media screen and (min-width: 992px) {
  .contCalendario {
    width: 900px;
  }
}
