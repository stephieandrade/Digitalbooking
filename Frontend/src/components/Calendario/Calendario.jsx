import React from "react";
import { Calendar } from "react-multi-date-picker";
import "./calendario.css";
import "./calendario-detalles.css";
import "./calendario-reserva.css";

export default function Calendario(props) {
  return (
    <>
      <div className="contCalendario">
        <Calendar {...props} />
      </div>
    </>
  );
}


