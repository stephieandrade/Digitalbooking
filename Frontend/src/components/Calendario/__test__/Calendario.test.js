import React from "react";
import { shallow } from "enzyme";
import Calendario from "../Calendario"

describe('Calendario Component', () =>{

    it("deber√≠a mostrar <Calendario/> correctamente ", () => {

    const wrapper = shallow(<Calendario/>);
     expect(wrapper).toMatchSnapshot();
    });

    it ('tiene un div', () => {

    const wrapper = shallow(<Calendario/>);
    const component = wrapper.find('div');
    expect(component.length).toEqual(1);

    });

    it('tiene una clase contCalendario', () => { 

    const wrapper = shallow(<Calendario/>);
    const node = wrapper.find('div');
     expect(node.hasClass('contCalendario')).toBeTruthy();
    })


});