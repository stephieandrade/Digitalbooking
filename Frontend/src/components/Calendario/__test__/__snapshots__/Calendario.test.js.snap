// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calendario Component deber√≠a mostrar <Calendario/> correctamente  1`] = `
<Fragment>
  <div
    className="contCalendario"
  >
    <ForwardRef(V) />
  </div>
</Fragment>
`;
